packages/netsuite-adapter/src/client/client.ts

```diff
@@ -203,6 +205,55 @@ export default class NetsuiteClient {
       .toArray()
   }
 
+  public static async createDependencyMap(
+    changes: ReadonlyArray<Change>,
+    deployReferencedElements: boolean,
+    elementsSourceIndex: LazyElementsSourceIndexes,
+  ):Promise<collections.map.DefaultMap<string, Set<string>>> {
+    const dependencyMap = new collections.map.DefaultMap<string, Set<string>>(() => new Set())
+    const elemIdsAndCustInfoArr = await awu(changes)
+      .map(getChangeData)
+      .map(async element => ({
+        elemId: element.elemID,
+        custInfos: await NetsuiteClient.toCustomizationInfos(
+          [element], deployReferencedElements, elementsSourceIndex
+        ),
+      })).toArray()
+
+    elemIdsAndCustInfoArr.forEach(elemIdAndCustInfo => {
+      elemIdAndCustInfo.custInfos.forEach(custInfo =>
+        lookupValue(custInfo.values, val => {
+          if (!_.isString(val)) {
+            return
+          }
+          const serviceIdInfoArray = captureServiceIdInfo(val)
+          serviceIdInfoArray.map(serviceIdInfo =>
+            dependencyMap
+              .get(elemIdAndCustInfo.elemId.getFullName())
+              .add(serviceIdInfo.serviceId))
+        }))
+    })
+    return dependencyMap
+  }
+
+  public static getFailedManifestErrorElemIds(
+    error: ManifestValidationError,
+    dependencyMap: collections.map.DefaultMap<string, Set<string>>,
+    changes: ReadonlyArray<Change>
+  ): Set<string> {
+    const changeData = changes.map(getChangeData)
+    log.debug('remove elements which contain a scriptid that doesnt exist in target account')
```

Comment:
maybe include `elementsToRemoveElemIDs` in the log message?
```suggestion
    log.debug('remove elements which contain a scriptid that doesnt exist in target account: %o', elementsToRemoveElemIDs)
```

Reply:
actually the log in line 303 covers it, so I think that log line can be removed.